### PR TITLE
SG-14619 python 3 port

### PR DESCRIPTION
**Pull request by pscadding**
Adds Python 3 support.

Also adds a box to ask if you would like to restart/reload your changes after performing config or core updates.
This prompt however is asked regardless of whether the updates completed successfully. I can remove it if we think it shouldn't be added.
I wonder if it might be better to instead of having a prompt have a button that sits next to the close dialog.
Shotgun__Check_for_core_updates_and_Comparing_master___SG-14619-Python-3-port_·_shotgunsoftware_tk-multi-devutils

This PR depends on updates to core, and will need the required core updating once they are released:
https://github.com/shotgunsoftware/tk-core/pull/766
https://github.com/shotgunsoftware/tk-core/pull/767

### CODE REVIEW

File: python/command_runner/interact.py [hunk 1]
```diff
@@ -73,12 +73,14 @@ def ask_yna_question(self, message, force_prompt=False):
...
-            QtGui.QMessageBox.Yes | QtGui.QMessageBox.No | QtGui.QMessageBox.YesToAll,
-        )
+        msgbx = QtGui.QMessageBox(self._parent)
+        msgbx.setWindowTitle("Shotgun")
+        msgbx.setIcon(msgbx.Icon.Question)
+        msgbx.setText(message)
+        msgbx.addButton(QtGui.QMessageBox.Yes)
+        msgbx.addButton(QtGui.QMessageBox.YesToAll)
+        msgbx.addButton(QtGui.QMessageBox.No)
+        res = msgbx.exec_()
```
Comment (pscadding) [inline python/command_runner/interact.py]: This was done, as I was seeing duplicate YesToAll buttons otherwise. I'm not sure why that was though.

Comment (pscadding) [reply python/command_runner/interact.py]: This has been fixed in core.

File: python/command_runner/dialog.py [hunk 2]
```diff
@@ -74,6 +75,12 @@ def _doit(self):
         self.ui.cancel.setText("Close")
         self.ui.cancel.clicked.connect(self.close)
 
+        reload = self._interaction_interface.ask_yn_question(
+            "Would you like to restart " "the engine to reload your changes?"
```
Comment (jfboismenu) [inline python/command_runner/dialog.py]: Why two strings?

**Review by jfboismenu — APPROVED** [review]
Looks good to me.

**Comment by jfboismenu** [comment]
Let's not worry too much about the UX here. This app is pretty obscure at the moment.

File: python/command_runner/dialog.py (hunk 2 above)
Comment (pscadding) [reply python/command_runner/dialog.py]: hmm good point, could it be that black rearranged things? Or I just messed up 😛

**Pull request merged**